File: SympleLang/src/Symple/Syntax/Token.cpp
#include "Symple/Common/Common.h"
#include "Symple/Syntax/Token.h"

namespace Symple
{
	Token::Token(TokenKind kind, std::string_view text)
		: kind(kind), text(text) {}

	TokenKind Token::GetKind() const
	{ return kind; }

	std::string_view Token::GetText() const
	{ return text; }
}

File: CommenSense/Token.cs
namespace CommenSense;

partial class Parser
{
	enum TokenKind
	{
		Unknown = -1,
		Eof,

		Str,
		Int,
		Float,
		Identifier,

		Plus,
		Minus,
		Star,
		Slash,
		Percent,
		SlashDash,
		PercentDash,

		And,

		Eql,
		DotDotDot,

		LeftBrace,
		RightBrace,
		LeftBracket,
		RightBracket,

		Comma,


		TrueKeyword,
		FalseKeyword,

		DeclKeyword,
	}

	record Token(TokenKind kind, string text);
}